Diagnostic/diagnostic.py

```diff
@@ -478,110 +483,125 @@ def install_service():
+                                          isSuccess=False,
+                                          version=hutil.get_extension_version(),
+                                          message="Install failed: " + msg)
+                sys.exit(error)  # Per extension specification, we must exit with a non-zero status code when install fails
+
+            if UseSystemdServiceManager:
                 install_service()
-                if error != 0:
-                    hutil.do_status_report("Install", "error", error, msg)
-                    sys.exit(error)
-                else:
-                    hutil.do_status_report("Install", "success",'0', "Install succeeded")
-            elif re.match("^([-/]*)(enable)", command):
-                hutil.exit_if_enabled()
+            hutil.do_status_report(ExtensionOperationType, "success", '0', "Install succeeded")
+
+        elif ExtensionOperationType is waagent.WALAEventOperation.Enable:
+            if UseSystemdServiceManager:
                 install_service()
-                RunGetOutput('systemctl enable mdsd-lde && systemctl restart mdsd-lde')
-                hutil.do_status_report("Enable","success",'0',"Enable succeeded")
-            elif re.match("^([-/]*)(daemon)", command):
-                start_mdsd()
-            elif re.match("^([-/]*)(update)", command):
-                hutil.do_status_report("Update", "success",'0', "Update succeeded")
-        else:
-            if re.match("^([-/]*)(disable)", command):
-                error,msg = stop_mdsd()
-                hutil.do_status_report("Disable", "success", '0',"Disable succeeded")
-            elif re.match("^([-/]*)(uninstall)", command):
-                stop_mdsd()
-                error,msg = uninstall_omi()
-                if EnableSyslog:
-                    error,msg = uninstall_rsyslogom()
-                hutil.do_status_report("Uninstall", "success",'0', "Uninstall succeeded")
-            elif re.match("^([-/]*)(install)", command):
-                error, msg = setup(True)
-                if error != 0:
-                    hutil.do_status_report("Install", "error", error, msg)
-                    sys.exit(error)
-                else:
-                    hutil.do_status_report("Install", "success",'0', "Install succeeded")
-            elif re.match("^([-/]*)(enable)", command):
+                RunGetOutput('systemctl enable mdsd-lde')
+                mdsd_lde_active = RunGetOutput('systemctl status mdsd-lde')[0] is 0
+                if not mdsd_lde_active or hutil.is_current_config_seq_greater_inused():
+                    RunGetOutput('systemctl restart mdsd-lde')
+            else:
                 # if daemon process not runs
-                hutil.log("get pid:"+str(get_mdsd_process()))
-                if len(get_mdsd_process())!=2 or  hutil.is_current_config_seq_greater_inused():
+                hutil.log("get pid:" + str(get_mdsd_process()))
```

Comment:
minor: get_mdsd_process() is called twice here. it's better to assign it to a var, then use the var in the 2nd case.